lib/smf/smf.c

```diff
@@ -9,24 +9,32 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(smf);
 
-/*
- * Private structure (to this file) used to track state machine context.
- * The structure is not used directly, but instead to cast the "internal"
- * member of the smf_ctx structure.
+/**
+ * @brief Private structure (to this file) used to track state machine context.
+ *        The structure is not used directly, but instead to cast the "internal"
+ *        member of the smf_ctx structure.
  */
 struct internal_ctx {
 	bool new_state : 1;
 	bool terminate : 1;
 	bool exit      : 1;
 	bool handled   : 1;
+#ifdef CONFIG_SMF_UML_HSM
+	bool entry     : 1;
+#endif
 };
 
-static bool share_paren(const struct smf_state *test_state,
-			const struct smf_state *target_state)
+/**
+ * @brief Test if any parent of the test state is the target state
+ *
+ * @param test_state   State to test if any parent is the target state
+ * @param target_state State being tested if it is a parent of the test state
+ * @return true if the target state is a parent of the test state
+ */
+__unused static bool share_paren(const struct smf_state *test_state,
```

Comment:
Thanks for adding header comments.  Do you mind splitting these and the formatting changes into a separate commit?